e2e/utils/pages/homepage.page.ts

```diff
@@ -217,11 +235,13 @@ export class HomepageLoggedOut extends Homepage {
 
 export class HomepageLoggedIn extends Homepage {
 	readonly #iiPage: InternetIdentityPage;
+	readonly promotionCarousel: PromotionCarousel;
 
 	constructor({ page, iiPage, viewportSize }: HomepageLoggedInParams) {
 		super({ page, viewportSize });
 
 		this.#iiPage = iiPage;
+		this.promotionCarousel = new PromotionCarousel(page);
```

Comment:
I think I did it the way I found it. iiPage is re-initiated aswell right? Or am I missing something? But in general you are correct, it could be just put into super. Then again we only need the carousel in homepageloggedin, since it wont display in loggedout

Reply:
no, what i meant is that it is already initiated in `super`, you are re-initiating it, no? isn't it a duplicated code?

Reply:
done